src/java/org/apache/cassandra/security/PEMBasedSslContextFactory.java

```diff
@@ -329,9 +406,9 @@ private KeyStore buildTrustStore() throws GeneralSecurityException, IOException
      * Enforces that the configuration specified a sole source of loading private keys - either {@code keystore} (the
      * actual file must exist) or {@code private_key}, not both.
      */
-    private void enforceSinglePrivateKeySource()
+    private void enforceSinglePrivateKeySource(boolean outbound)
     {
-        if (keystoreFileExists() && !StringUtils.isEmpty(pemEncodedKey))
+        if (keystoreFileExists(outbound) && !StringUtils.isEmpty(pemEncodedKey))
         {
             throw new IllegalArgumentException("Configuration must specify value for either keystore or private_key, " +
                                                "not both for PEMBasedSSlContextFactory");
```

Comment:
The condition was wrong. It checks the same `pemEncodedKey` for both inbound and outbound. Please using the following and remove `keystoreFileExists`

```suggestion
    private void enforceSinglePrivateKeySource()
    {
        if (keystoreContext.hasKeystore() && !StringUtils.isEmpty(pemEncodedKey)
            || outboundKeystoreContext.hasKeystore() && !StringUtils.isEmpty(pemEncodedOutboundKey))
        {
            throw new IllegalArgumentException("Configuration must specify value for either keystore or private_key, " +
                                               "not both for PEMBasedSSlContextFactory");
```